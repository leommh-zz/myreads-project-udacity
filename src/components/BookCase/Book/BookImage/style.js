const style = {
    cover: {
        width: 128,
        height: 193,
    },
};

export default style;