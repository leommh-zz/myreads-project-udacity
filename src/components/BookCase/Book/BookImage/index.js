import React from 'react';
import { CardMedia } from '@material-ui/core';
import style from './style';

const BookImage = ({ images='' }) => {
    const thumbnail = images.thumbnail !== undefined ? images.thumbnail : 'https://i.imgur.com/GZYl7cN.png';

    return (
        <CardMedia
            style={style.cover}
            image={thumbnail}    
            src="img"                    
        /> 
    )
}

export default BookImage;