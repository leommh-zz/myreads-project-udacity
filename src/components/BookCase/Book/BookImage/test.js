import React from 'react';
import { shallow } from 'enzyme';
import BookImage from './index';

describe('[Component] BookImage', () => {

    const wrapper = shallow(<BookImage image='' />);

    it('Display BookImage Component with props', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
});