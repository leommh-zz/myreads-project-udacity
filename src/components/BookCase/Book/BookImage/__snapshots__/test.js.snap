// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] BookImage Display BookImage Component with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookImage
    image=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "image": "https://i.imgur.com/GZYl7cN.png",
      "src": "img",
      "style": Object {
        "height": 193,
        "width": 128,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "image": "https://i.imgur.com/GZYl7cN.png",
        "src": "img",
        "style": Object {
          "height": 193,
          "width": 128,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
